src/components/ProductList/ProductList.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useState } from 'react';
+import {
+  AsideBar,
+  AsideItem,
+  ImageGrid,
+  ImageResult,
+  ProductListMain, 
+  Spiner} from './ProductList.styles';
+import CategoriesData from '../../mocks/en-us/product-categories.json';
+import FeaturedProducts from '../../mocks/en-us/featured-products.json';
+import { ImageWrap, InfoImage } from '../Home/FeaturedProducts/FeaturedProducts.styled';
+import { PaginationControl } from './PaginationControl/PaginationControl';
+
+export const ProductList = () => {
+  const [results, setResults] = useState([]);
+  const [filters, setFilters] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  let filtersSet = [];
+
+  useEffect(() => {
+    setIsLoading(true);
+    setTimeout(() => {
+      const allItems = createCards(FeaturedProducts.results);
+      setResults(allItems);
+      setIsLoading(false);
+    }, 2000);
+  }, []);
+
+
+  function handleClick(categoryName) {
+    filtersSet = manageCategories(categoryName, filters);
+    setFilters(filtersSet);
+    let cards = []
```

Comment:
const cardsToFilter = !filtersSet.length ?FeaturedProducts.results: filterProducts(filtersSet)
const cards = createCards(cardsToFilter);
setResults(cards);


what do you think?

Reply:
Very nice, cleaner. Thank you!